### PR TITLE
More stable pattern

### PR DESCRIPTION
**Pull request by lippoliv**
Hey there,
I mentioned some trouble with the current replace-code.
I got a broken match on Images at the very end of an article and some global footer plugins wich also hooked into `the_content` Filter. The Text was not placed beside the Image, it was far away.

Changing the `old_content` match helped me, I did not see any side-effect.

### CODE REVIEW

**Comment by webzunft** [comment]
@lippoliv thank you! I am not ready to change this completely, but maybe a filter would work too.
Could you please post the output of the full "matches" array here so I can compare it?
Thanks

**Comment by lippoliv** [comment]
@webzunft sorry for the late reply (I moved to a new home).
I'am not really sure what you do want from me...

Here's the "real" Problem

image

**Comment by lippoliv** [comment]
OK @webzunft I saw that I did a mistake. Your PHP Looks OK, but your regex is not precise enough.

I broke down one good HTML (from the screenshot above) removed all useless TEXT and here you have to examples:

1. the current one https://regex101.com/r/LM0FcN/2, you see the match is across multiple Image tags.
2. my last modification from f8e73a8 https://regex101.com/r/LM0FcN/1, the hit is just the one Image that should be modified. With that Expression the "arrow from the screenshot above" will be reallized.

At the moment I see no problem with my change. What I did is just replace the `.*` with `[^>]*` and `[^"]*` to Keep the expression within the element definition and dont allow him to leave the closing braket of the IMG or the closing paragraph of the CLASS-Attribute.

I checked three known blog-posts on our Plattform wich had wrong-placed source-texts and all are working good now.

Let me know if you need me to do further Tests.

**Comment by webzunft** [comment]
Thanks, @lippoliv. Seems stable and very useful indeed.

File: public/public.php [hunk 1]
```diff
@@ -65,7 +65,7 @@ public function content_filter($content)
             // display inline sources
             $options = $this->get_isc_options();
             if (isset($options['display_type']) && is_array($options['display_type']) && in_array('overlay', $options['display_type'])) {
-                $pattern = '#(\[caption.*align="(.+)"[^\]*]{0,}\])? *(<a [^>]+>)? *(<img .*class=".*(alignleft|alignright|alignnone|aligncenter)??.*wp-image-(\d+)\D*".*src="(.+)".*/?>).*(?(3)(?:</a>)|.*).*(?(1)(?:\[/caption\])|.*)#isU';
+                $pattern = '#(\[caption.*align="(.+)"[^\]*]{0,}\])? *(<a [^>]+>)? *(<img [^>]*class="[^"]*(alignleft|alignright|alignnone|aligncenter)??[^"]*wp-image-(\d+)\D*"[^>]*src="(.+)".*/?>).*(?(3)(?:</a>)|.*).*(?(1)(?:\[/caption\])|.*)#isU';
```
Comment (goatherd) [inline public/public.php]: Just a quick note-dump:

The regex misses single quoted tags, has uncertain unicode support (`u`-modifier recommended and usage of unicode-classes like `\pL` when really needed) and has some other hard coded byte dependencies like spaces and un-empty `<a>`-tags.

Greediness-settings and matching seems a bit off performance-wise (may want to use `(?:..)` for non-matching parenthesis; use named parenthesis or constant numbering. Naming magic numbers is plain weird to my understanding).

**Pull request merged**